configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/FileDistributionImpl.java

```diff
@@ -10,26 +9,26 @@
 import com.yahoo.jrt.StringArray;
 import com.yahoo.jrt.Supervisor;
 import com.yahoo.jrt.Target;
-import java.util.logging.Level;
-import com.yahoo.vespa.defaults.Defaults;
+import com.yahoo.jrt.Transport;
 
 import java.io.File;
 import java.util.Set;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
  * @author baldersheim
  */
 public class FileDistributionImpl implements FileDistribution, RequestWaiter {
+
     private final static Logger log = Logger.getLogger(FileDistributionImpl.class.getName());
     private final static double rpcTimeout = 1.0;
 
-    private final Supervisor supervisor;
+    private final Supervisor supervisor = new Supervisor(new Transport("filedistribution"));
```

Comment:
You need to close/stop the supervisor too, when you create it. It contains threads.

Reply:
FileDistributionFactory is a singleton and lives for as long as the process is running, I guess that's why it is not closed.